File: frontend/src/entity/Email.js
import { baseFromFirestore, omitUndefined } from './_helpers.js'

export const defaultShape = {
  id: null,
  uid: '',
  to: '',
  subject: '',
  body: '',
  status: 'pending',
  sendAt: null,
  sentAt: null,
  createdAt: null,
}

export function toFirestore(data) {
  const { id: _id, ...rest } = { ...defaultShape, ...data }
  return omitUndefined(rest)
}

export function fromFirestore(snapshot) {
  const doc = baseFromFirestore(snapshot)
  if (!doc) return null
  return { ...defaultShape, ...doc }
}

File: backend/entity/email.py
from datetime import datetime

from pydantic import BaseModel


class EmailModel(BaseModel):
    id: str | None = None
    uid: str
    to: str
    subject: str
    body: str
    status: str = "pending"
    send_at: datetime | None = None
    sent_at: datetime | None = None


class SendEmailRequest(BaseModel):
    to: str
    subject: str
    body: str


class ScheduleEmailRequest(BaseModel):
    to: str
    subject: str
    body: str
    send_at: datetime
